driver/SDL/sdl_common.c

```diff
@@ -74,22 +82,12 @@ void sdl_keyboard_read(lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
 {
     (void) indev_drv;      /*Unused*/
 
-    static bool dummy_read = false;
-    const size_t len = strlen(buf);
-
-    /*Send a release manually*/
-    if (dummy_read) {
-        dummy_read = false;
-        data->state = LV_INDEV_STATE_RELEASED;
-        data->continue_reading = len > 0;
-    }
-        /*Send the pressed character*/
-    else if (len > 0) {
-        dummy_read = true;
-        data->state = LV_INDEV_STATE_PRESSED;
-        data->key = buf[0];
-        memmove(buf, buf + 1, len);
+    if(key_buffer_len > 0){
+        data->state = key_buffer[0].state;
+        data->key = key_buffer[0].key;
         data->continue_reading = true;
```

Comment:
`continue_reading` should be set to `true` only if the fifo is not empty after popping current key.  
Here you always set it to `true`.

Reply:
```c
        data->continue_reading = key_buffer_len > 0;
        key_buffer_len--;
```
Don't you need to decrement `key_buffer_len` **before** setting `data->continue_reading` according to it?

Reply:
Yes you're right. It won't get to this line of code in this order and therefore never set continue_reading to false.